src/Crawler.php

```diff
@@ -459,21 +463,25 @@ protected function createRobotsTxt(UriInterface $uri): RobotsTxt
 
     protected function getCrawlRequests(): Generator
     {
-        while ($crawlUrl = $this->crawlQueue->getFirstPendingUrl()) {
-            if (! $this->crawlProfile->shouldCrawl($crawlUrl->url)) {
-                $this->crawlQueue->markAsProcessed($crawlUrl);
-
-                continue;
-            }
-
-            if ($this->crawlQueue->hasAlreadyBeenProcessed($crawlUrl)) {
+        while (
+            $this->maximumCrawlCountReached() === false &&
+            $crawlUrl = $this->crawlQueue->getFirstPendingUrl()
+        ) {
+            // Skip any undesired or already processed URLs.
```

Comment:
Please remove these new comments, I think the code is clear enough

Reply:
Done @freekmurze